cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ArtifactHttpHandlerInternal.java

```diff
@@ -107,14 +107,24 @@ public void getArtifactBytes(HttpRequest request, HttpResponder responder,
                                @PathParam("namespace-id") String namespaceId,
                                @PathParam("artifact-name") String artifactName,
                                @PathParam("artifact-version") String artifactVersion,
-                               @QueryParam("scope") @DefaultValue("user") String scope) throws Exception {
+                               @QueryParam("scope") @DefaultValue("user") String scope,
+                               @QueryParam("lastModified") @DefaultValue("0") String lastModified) throws Exception {
```

Comment:
Why use query param?? We should be using the standard HTTP `If-Modified-Since` request header.

Reply:
done